ui/admin/tests/e2e/helpers/boundary-ui.js

```diff
@@ -287,3 +287,140 @@ exports.addBrokeredCredentialsToTarget = async (
   await page.getByRole('button', { name: 'Dismiss' }).click();
   await expect(page.getByRole('link', { name: credentialName })).toBeVisible();
 };
+
+/**
+ * Uses the UI to create new Auth Method. Assumes you have selected the desired scope.
+ * @param {Page} page Playwright page object
+ * @param {string} authMethodName Name of new auth method
+ */
+exports.createNewPasswordAuthMethod = async (page, authMethodName) => {
+  await page
+    .getByRole('navigation', { name: 'IAM' })
+    .getByRole('link', { name: 'Auth Methods' })
+    .click();
+  await page.getByTitle('New', { exact: true }).click();
+  await page.getByText('Password', { exact: true }).click();
+  await page.getByLabel('Name').fill(authMethodName);
+  await page.getByRole('button', { name: 'Save' }).click();
+  await expect(
+    page.getByRole('alert').getByText('Success', { exact: true })
+  ).toBeVisible();
+  await page.getByRole('button', { name: 'Dismiss' }).click();
+  await expect(
+    page
+      .getByRole('navigation', { name: 'breadcrumbs' })
+      .getByRole('link', { name: authMethodName })
+  ).toBeVisible();
+};
+
+/**
+ * Uses the UI to make the first available auth method primary.
+ * Assumes you have created new auth method.
+ * @param {Page} page Playwright page object
+ */
+exports.makeAuthMethodPrimary = async (page) => {
+  await page
+    .getByRole('navigation', { name: 'IAM' })
+    .getByRole('link', { name: 'Auth Methods' })
+    .click();
+  await page.click('[aria-label="Manage"]');
```

Comment:
Would a simpler locator work better here?
```
await page.getByText('Manage').click();
```
I want to try to avoid using attributes when possible.

Reply:
Agreed. Interesting behaviour:
This one works:
`page.getByRole("button", { name: 'Manage' }) `

This one does not:
`page.getByText('Manage')`

Reply:
That's fine. Were there other instances of the word `Manage` on the page?

Reply:
There is one if there is one auth method created as it is in the test.
When using `page.getByText('Manage')`, it keeps waiting for the element but could not find it as 'Manage' is in the`area-label` property only.